### PR TITLE
NT-2012:Finalize the Adoption of View Bindings

### PR DESCRIPTION
**Pull request by hadia**
# 📲 What

Remove the use of Kotlin Synthetics

# 🤔 Why

As its deprecated 

# 🛠 How

- Remove Dependence  in build Gradle
- Refactor all views

> 
> ### - Activities  [ ] 
> 
> 1. Account activity
> 2. Login activity
> 3. UpdateActivity 
> 4. EditProfileActivity 
> 5. SettingsActivity 
> 6. ProfileActivity.kt 
> 7. ResetPasswordActivity 
> 8. ChangePasswordActivity 
> 9. ChangeEmailActivity
> 10. PrivacyActivity
> 11. CampaignDetailsActivity
> 12. MessageCreatorActivity
> 13. PaymentMethodsSettingsActivity
> 14. NotificationsActivity
> 15. NewsletterActivity
> 16. CreatorBioActivity
> 17. CreatePasswordActivity
> 18. EditorialActivity
> 19. ProjectActivity
> 20. FeatureFlagsActivity
> 
> ### - Fragments  [ ] 
> 
> 1. RewardsFragment
> 2. BackingAddOnsFragment
> 3. CancelPledgeFragment
> 4. DiscoveryFragment
> 5. NewCardFragment
> 6. PledgeFragment
> 
> ### - Custom Views  [ ] 
> 
> 1. WebView
> 2. RewardCardUnselectedViewHolder
> 3. ShippingRuleViewHolder

# 📋 QA
need to test all changed views and activities

# Story 📖

https://kickstarter.atlassian.net/browse/NT-2012

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -13,7 +13,7 @@ apply plugin: 'com.google.firebase.crashlytics'
 apply plugin: 'com.google.firebase.firebase-perf'
 apply plugin: 'kotlin-android'
 apply plugin: 'kotlin-kapt'
-apply plugin: 'kotlin-android-extensions'
```
Comment (Arkariang) [inline app/build.gradle:16]: ❤️

File: app/src/internal/java/com/kickstarter/ui/activities/FeatureFlagsActivity.kt [hunk 2]
```diff
@@ -1,23 +1,17 @@
 package com.kickstarter.ui.activities
```
Comment (Arkariang) [inline app/src/internal/java/com/kickstarter/ui/activities/FeatureFlagsActivity.kt:1]: Tested on my environment, This activity looks good to me
Screen Shot 2021-08-06 at 3 41 40 PM

File: app/src/main/java/com/kickstarter/ui/activities/AccountActivity.kt [hunk 3]
```diff
@@ -7,15 +7,14 @@ import android.widget.AdapterView
 import android.widget.ArrayAdapter
```
Comment (Arkariang) [inline app/src/main/java/com/kickstarter/ui/activities/AccountActivity.kt:7]: Tested on my environment with use cases:
- Change password: OK
- Change Currentcy: Ok
- Add a new payment method: Ok

File: app/src/main/java/com/kickstarter/ui/activities/CampaignDetailsActivity.kt [hunk 4]
```diff
@@ -30,14 +32,16 @@ class CampaignDetailsActivity : BaseActivity<CampaignDetailsViewModel.ViewModel>
...
-            .subscribe { ViewUtils.setGone(campaign_details_pledge_container, !it) }
+            .subscribe {
+                binding.campaignDetailsPledgeContainer.isGone = !it
+            }
 
         this.viewModel.outputs.url()
             .observeOn(AndroidSchedulers.mainThread())
             .compose(bindToLifecycle())
-            .subscribe { web_view.loadUrl(it) }
+            .subscribe { binding.webView.loadUrl(it) }
```
Comment (Arkariang) [inline app/src/main/java/com/kickstarter/ui/activities/CampaignDetailsActivity.kt:42]: The campaign details has stop loading, attache video comparing latest master behaviour VS this branch behaviour

https://user-images.githubusercontent.com/4083656/128578693-db532681-fa9b-45c8-88de-6ca172fe4978.mp4

File: app/src/main/java/com/kickstarter/ui/activities/ChangeEmailActivity.kt [hunk 5]
```diff
@@ -7,62 +7,65 @@ import android.view.MenuItem
...
-import kotlinx.android.synthetic.main.activity_change_email.*
-import kotlinx.android.synthetic.main.change_email_toolbar.*
 import rx.android.schedulers.AndroidSchedulers
 
 @RequiresActivityViewModel(ChangeEmailViewModel.ViewModel::class)
 class ChangeEmailActivity : BaseActivity<ChangeEmailViewModel.ViewModel>() {
 
     private var saveEnabled = false
 
+    private lateinit var binding: ActivityChangeEmailBinding
```
Comment (Arkariang) [inline app/src/main/java/com/kickstarter/ui/activities/ChangeEmailActivity.kt:25]: Something off with the ChangeEmailActivity, on the lates master I can request the email change and shows a UI saying a verification email has been sent. But in this branch it tells me the password is incorrect and do not let me change the email.

File: app/src/main/java/com/kickstarter/ui/activities/ChangePasswordActivity.kt [hunk 6]
```diff
@@ -4,44 +4,48 @@ import android.content.Intent
 import android.os.Bundle
```
Comment (Arkariang) [inline app/src/main/java/com/kickstarter/ui/activities/ChangePasswordActivity.kt:4]: Working fine for me 👍

File: app/src/main/java/com/kickstarter/ui/activities/CreatorBioActivity.kt [hunk 7]
```diff
@@ -4,35 +4,39 @@ import android.content.Intent
 import android.os.Bundle
```
Comment (Arkariang) [inline app/src/main/java/com/kickstarter/ui/activities/CreatorBioActivity.kt:4]: Creator Bio not loading, but on latest master it is loading

https://user-images.githubusercontent.com/4083656/128579689-67118677-7d29-4f4b-b51f-8ed4912ecc15.mp4

File: app/src/main/java/com/kickstarter/ui/activities/LoginActivity.kt [hunk 8]
```diff
@@ -5,6 +5,7 @@ import android.content.Intent
 import android.os.Bundle
```
Comment (Arkariang) [inline app/src/main/java/com/kickstarter/ui/activities/LoginActivity.kt:5]: Working perfectly
The only scenario I haven't test has been log in with facebook, the other user paths works perfectly, even the 2fa

File: app/src/main/java/com/kickstarter/ui/activities/ProjectActivity.kt [hunk 9]
```diff
@@ -21,6 +21,7 @@ import androidx.fragment.app.FragmentManager
 import androidx.recyclerview.widget.LinearLayoutManager
 import com.google.firebase.crashlytics.FirebaseCrashlytics
```
Comment (Arkariang) [inline app/src/main/java/com/kickstarter/ui/activities/ProjectActivity.kt:22]: Project Activity works fine on my environment, Except the Campaign description and creator bio not loading.
- Like button: Ok
- Share button: ok
- Pledge Flow: Ok
- Manager pledge Flow: Ok
- Updates: Ok
- Comments: Ok

File: app/src/internal/java/com/kickstarter/ui/activities/FeatureFlagsActivity.kt (hunk 2 above)
Comment (hadia) [reply app/src/internal/java/com/kickstarter/ui/activities/FeatureFlagsActivity.kt:1]: 😍

File: app/src/main/java/com/kickstarter/ui/activities/AccountActivity.kt (hunk 3 above)
Comment (hadia) [reply app/src/main/java/com/kickstarter/ui/activities/AccountActivity.kt:7]: 👌

File: app/src/main/java/com/kickstarter/ui/activities/CampaignDetailsActivity.kt (hunk 4 above)
Comment (hadia) [reply app/src/main/java/com/kickstarter/ui/activities/CampaignDetailsActivity.kt]: Fixed
https://user-images.githubusercontent.com/1075310/128724440-1844eef3-e812-4b62-b026-7cf2ecde1490.mp4

File: app/src/main/java/com/kickstarter/ui/activities/ChangeEmailActivity.kt (hunk 5 above)
Comment (hadia) [reply app/src/main/java/com/kickstarter/ui/activities/ChangeEmailActivity.kt:25]: Fixed

File: app/src/main/java/com/kickstarter/ui/activities/CreatorBioActivity.kt (hunk 7 above)
Comment (hadia) [reply app/src/main/java/com/kickstarter/ui/activities/CreatorBioActivity.kt:4]: Fixed as the issue was in web view

File: app/src/main/java/com/kickstarter/ui/adapters/ShippingRulesAdapter.kt [hunk 10]
```diff
@@ -42,8 +45,8 @@ class ShippingRulesAdapter(ctx: Context, private val resourceId: Int, val items:
         }
     }
 
-    override fun viewHolder(layout: Int, view: View): KSArrayViewHolder {
```
Comment (Arkariang) [inline app/src/main/java/com/kickstarter/ui/adapters/ShippingRulesAdapter.kt:45]: Something off on the Shipping Rule Selector.
Attached video comparing it with the latest master

https://user-images.githubusercontent.com/4083656/128768361-d3b440f6-376d-4915-b1ba-6d63013d1442.mp4

**Review by Arkariang — APPROVED** [review]
lgtm! great job @hadia I've tested all possible scenario that came to my mind

File: app/src/main/java/com/kickstarter/ui/fragments/PledgeFragment.kt [hunk 11]
```diff
@@ -21,19 +21,23 @@ import android.widget.TextView
 import androidx.annotation.NonNull
 import androidx.constraintlayout.widget.ConstraintSet
 import androidx.core.content.ContextCompat
+import androidx.core.view.isGone
```
Comment (Arkariang) [inline app/src/main/java/com/kickstarter/ui/fragments/PledgeFragment.kt:24]: Tested 
- Pledge adding a new card in the pledge : ok
- Pledge using and existing card : ok
- Pledge with add-ons with shipping: ok
- Update pledge without add ons: ok
- Cancel pledge: ok
- Change payment method without addons: ok
- Change reward: ok

**Pull request merged**